### ISSUE TITLE
🐛 [Bug Report] Shadow Word: Death, Judgement of Blood and more

### ISSUE THREAD

**Issue by ST369** [cmangos/issues#3062]
### Bug Details
1) Shadow Word: Death – backlash damage is delayed, but target damage is not (no spell batching).
This leads to the scenario where the target's spell is delayed more than the backlash damage, so it's not possible to break crowd control like Polymorph.
Another problem is that damage absorbed by the target mirrors backlash damage, meaning SWD is ineffective in breaking crowd control if the damage is fully absorbed.
2) Judgement of Blood – neither target damage nor backlash has delay. Same problem with absorbed damage.
3) The self-damage effects of other similar abilities used to break crowd control are also not delayed due to the lack of spell batching. Bloodrage, Life Tap, Dark/Demonic Rune, Hellfire, etc.
### Expected behavior
1) "The backlash the caster will take always equals that of the target, even if the damage was zero. Damage that is partially resisted does not backlash; absorbed damage does, however." [wowpedia rev: 2008-8-24](https://wowpedia.fandom.com/wiki/Shadow_Word:_Death?oldid=1583669)
2) [satanz ret - seal of blood [1:38]](https://www.youtube.com/watch?v=LgNguOcsuhM#t=1m38s)
[satanz ret - seal of blood [0:23]](https://www.youtube.com/watch?v=LgNguOcsuhM&t=23s)
3) [Diary of a Corruptor [8:37]](https://youtu.be/yU6T3RA77ZE?t=517)
"Sorrow's Polymorph is cleansed by Sorrow's Hellfire."
https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-exploits/145833-break-polymorph-warrior.html
https://us.forums.blizzard.com/en/wow/t/spell-batching-polymorph-with-bloodrage/873755
### Suggested Workaround
Add artificial delay to these abilities, on both target and self damage, so that the behavior more closely resembles the original game.
### Core SHA1 Commit Hash
852785df69a2fc504f08d3ed1224db5c12d4e9ce
### Database SHA1 Commit Hash
185c1695c0a6a4040ffbe5edb77bfc3df1aee762
### Operating System
Windows 10
### Client Version
2.4.3 (The Burning Crusade)

**Comment by killerwife** [comment]
This is intended as of now. Spell batching has many development downsides and as of right now we do not plan to implement it (lack of actually understanding is one of the reasons). The system also redirects casting in a non trivial manner and going forward for now its easier to maintain core without it. Adding artificial delays is not the way to go, because even that will not reproduce the side effects you describe. You actually need to somewhat snapshot some effects during queue execution, so that you do not introduce unblizzlike side effects on their own.

TL;DR

Until we are more "done" with the core, its more trouble than its worth. And it also makes the game feel terribly sluggish.

I am the resident spell developer and even I come nowhere close to be able to implement a proper spell queue. Closing this for now, but it can be revisited in the future.
